### DISCUSSION TITLE
niswitch session termination

### DISCUSSION THREAD

**Discussion by azviu** [ni/nimi-python#1895] [Q&A]
Is there anyway to list out the un-terminated niswitch session? It is useful in cases when another application created a session for the PXI switch and quit unexpectedly. It can only be terminated if the session info is known.

**Comment by marcoskirsch** [comment]
> It is useful in cases when another application created a session for the PXI switch and quit unexpectedly. It can only be terminated if the session info is known.

Can you provide more detail on what you are observing?

What you describe is not how the NI-SWITCH driver works. If an application creates a session to a PXI switch and exits without closing the session, either gracefully or not, the driver cleans up and there is no session that the user needs to terminate. Just open a new session normally.

**Reply by azviu** [reply]
hi Marco, thanks for getting back to me. You're correct. End up I found out there are couple applications have session created for the switch. Once those applications quit. the session is released I'm able to start a new session again.
